src/argilla/server/contexts/datasets.py

```diff
@@ -723,13 +737,17 @@ async def _build_record_suggestions(
     return suggestions
 
 
+R = TypeVar("R")
+
+
 async def _build_record_vectors(
     db: "AsyncSession",
     dataset: Dataset,
     record_schema: Union["RecordCreate", "RecordUpdateWithId"],
```

Comment:
We're using `RecordUpdate` schemas in from the `update_record` method
```suggestion
    record_schema: Union["RecordCreate", "RecordUpdate", "RecordUpdateWithId"],
```